### PR TITLE
PW-31 Add testing methodology

### PR DESCRIPTION
**Pull request by KToldy**
Add testing methodology + output pdf file

### CODE REVIEW

File: src/metodiky/metodika-testovania.tex [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        \textnormal{%
+     Unit testy umožňujú testovať individuálne jednotky kódu nezávisle od ostatných. Za jednotku považujeme samostatne testovateľnú časť programu, napr. funkciu, či jednotlivé komponenty. \\\\
+Unit testy by mali byť písané tak, aby overovali práve jednu funkcionalitu. Ak sa unit test skladá z viacerých častí, ktoré sa dajú dekomponovať, je ich potrebné rozdeliť do viacerých unit testov.\\\\
+Unit testy využívajú framework Mocha spolu s knižnicou Chai. V unit teste porovnávame vopred špecifikované hodnoty s výstupmi testovanej funkcie alebo metódy.\\\\
+Unit testy sú uložené v adresári \/tests\/unit, v ktorom je na každý komponent vytvorený samostatný test. Pomenovanie jednotlivých testov musí dodržiavať menovaciu konvenciu *.spec.js. 
+Testy je možné spustiť príkazom:
+npm run test:unit
+        }
+				
+				   \section*{End to End testy \(E2E\)}
```
Comment (MCFreddie777) [inline src/metodiky/metodika-testovania.tex]: "E2E Testy" is sufficient.

File: src/metodiky/metodika-testovania.tex [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    \include{title-page}
+
+    \section*{Úvod}
+
+        \textnormal{%
+        Metodika testovania definuje typy testov, spôsob ich písania a spúšťania. Testovanie softvéru a jeho častí je dôležitou súčasťou vývoja softvéru. Akákoľvek, hoci aj malá zmena môže ovplyvniť funkcionalitu súvisiacich komponentov, preto je potrebné aby bola správnosť danej funkcionality vždy pred jej schválením otestovaná. \\\\
+Každá nová funkcionalita systému, ktorá bola implementovaná, musí prejsť testovaním. Testy sa vyvárajú jednotkové, ale aj integračné, pričom testy musia pokrývať aj určitú časť funkcionality.
+        }
+		\section*{UNIT testy}
```
Comment (MCFreddie777) [inline src/metodiky/metodika-testovania.tex]: No need for caps... Just `Unit testy`

File: src/metodiky/metodika-testovania.tex [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+        \textnormal{%
+     Unit testy umožňujú testovať individuálne jednotky kódu nezávisle od ostatných. Za jednotku považujeme samostatne testovateľnú časť programu, napr. funkciu, či jednotlivé komponenty. \\\\
+Unit testy by mali byť písané tak, aby overovali práve jednu funkcionalitu. Ak sa unit test skladá z viacerých častí, ktoré sa dajú dekomponovať, je ich potrebné rozdeliť do viacerých unit testov.\\\\
+Unit testy využívajú framework Mocha spolu s knižnicou Chai. V unit teste porovnávame vopred špecifikované hodnoty s výstupmi testovanej funkcie alebo metódy.\\\\
+Unit testy sú uložené v adresári \/tests\/unit, v ktorom je na každý komponent vytvorený samostatný test. Pomenovanie jednotlivých testov musí dodržiavať menovaciu konvenciu *.spec.js.
```
Comment (MCFreddie777) [inline src/metodiky/metodika-testovania.tex:31]: seems like the escaping did not work in the result doc

**Review by MCFreddie777 — DISMISSED** [review]
Left some comments, otherwise LGTM.

**Review by MaliniakovaKarin — DISMISSED** [review]

File: src/metodiky/metodika-testovania.tex [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+Unit testy sú uložené v adresári \/tests\/unit, v ktorom je na každý komponent vytvorený samostatný test. Pomenovanie jednotlivých testov musí dodržiavať menovaciu konvenciu *.spec.js. 
+Testy je možné spustiť príkazom:
+npm run test:unit
+        }
+				
+				   \section*{End to End testy \(E2E\)}
+
+        \textnormal{%
+        Metodika testovania definuje typy testov, spôsob ich písania a spúšťania. Testovanie softvéru a jeho častí je dôležitou súčasťou vývoja softvéru. Akákoľvek, hoci aj malá zmena môže ovplyvniť funkcionalitu súvisiacich komponentov, preto je potrebné aby bola správnosť danej funkcionality vždy pred jej schválením otestovaná. \\\\
+Každá nová funkcionalita systému, ktorá bola implementovaná, musí prejsť testovaním. Testy sa vyvárajú jednotkové, ale aj integračné, pričom testy musia pokrývať aj určitú časť funkcionality.
```
Comment (MaliniakovaKarin) [inline src/metodiky/metodika-testovania.tex]: "vyvárajú" change "vytvárajú"

File: src/metodiky/metodika-testovania.tex [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        }
+				
+				   \section*{Postup testovania}
+
+        \textnormal{%
+\begin{enumerate}
+\item Napísanie kódu, ktorý sa bude testovať.
```
Comment (MaliniakovaKarin) [inline src/metodiky/metodika-testovania.tex]: change sentences like this: "Napíš kód, ktorý sa bude testovať"

**Review by MaliniakovaKarin — DISMISSED** [review]

**Review by MaliniakovaKarin — APPROVED** [review]

**Pull request merged**